### PR TITLE
ENH - rewrite rally_extract_tests.py

### PR DESCRIPTION
**Pull request by anish-mudaraddi**
rewrite rally_extract_tests.py incorporating changes from #89 
Needs unit-tests and proper testing

### CODE REVIEW

File: OpenStack-Rally-Tester/usr/local/bin/rally_extract_results.py [hunk 1]
```diff
@@ -4,94 +4,217 @@
...
+
+    post_to_influxdb(
+        build_datastring(metrics, config["cloud.instance"]),
+        host=config["db.host"],
+        db_name=config["db.database"],
+        auth=(config["auth.username"], config["auth.password"]),
+    )
+
+
+def build_datastring(metrics: List, instance: str):
```
Comment (khalford) [inline OpenStack-Rally-Tester/usr/local/bin/rally_extract_results.py:42]:
```suggestion
def build_datastring(metrics: List, instance: str) -> str:
```

File: OpenStack-Rally-Tester/usr/local/bin/rally_extract_results.py [hunk 2]
```diff
@@ -4,94 +4,217 @@
...
-        datastring += (
-            metric["measurement"].replace(".", "-")
-            + ",instance="
-            + metric["tags"]["instance"]
-            + ",workhours="
-            + str(workhours)
+from datetime import datetime
+
+
+def rally_extract_results(report_path: Path, config_path: Path):
```
Comment (khalford) [inline OpenStack-Rally-Tester/usr/local/bin/rally_extract_results.py:18]:
```suggestion
def rally_extract_results(report_path: Path, config_path: Path) -> None:
```

File: OpenStack-Rally-Tester/usr/local/bin/rally_extract_results.py [hunk 3]
```diff
@@ -4,94 +4,217 @@
...
+
+def parse_entry_single_result(test_entry: Dict) -> Dict:
+    """
+    generates metrics from a rally report entry that contains single result
+    :param test_entry: a rally report entry that contains a single result entry
+    :return: a set of metrics for that rally report entry
+    """
+    success_flag = 1
+    if test_entry["sla"]:
+        success_flag = int(all(sla["success"] for sla in test_entry["sla"]))
```
Comment (khalford) [inline OpenStack-Rally-Tester/usr/local/bin/rally_extract_results.py:115]:
```suggestion
    if test_entry["sla"]:
        temp_list = [sla["success"] for sla in test_entry["sla"]]
        success_flag = int(all(temp_list))
```

File: OpenStack-Rally-Tester/usr/local/bin/rally_extract_results.py [hunk 4]
```diff
@@ -4,94 +4,217 @@
...
+    required_values = [
+        "auth.password",
+        "auth.username",
+        "cloud.instance",
+        "db.database",
+        "db.host",
+    ]
+    assert all(
+        val in config_dict for val in required_values
+    ), "Config file is missing required values."
```
Comment (khalford) [inline OpenStack-Rally-Tester/usr/local/bin/rally_extract_results.py:173]: Do we want to error with which values we are missing here?

File: OpenStack-Rally-Tester/usr/local/bin/rally_extract_results.py [hunk 5]
```diff
@@ -4,94 +4,217 @@
...
+    try:
+        read_config_file(config_path)
+    except configparser.Error as exp:
+        raise RuntimeError(
+            f"could not read influx db config file '{args.config_filepath}'"
+        ) from exp
+
+    rally_extract_results(args.rally_filepath, config_path)
+
+
```
Comment (khalford) [inline OpenStack-Rally-Tester/usr/local/bin/rally_extract_results.py:218]:
```suggestion
    check_args(args)
    rally_extract_results(args.rally_filepath, config_path)

def check_args(args;  argparse.parser) -> None:
    try:
        args = parser.parse_args(user_args)
    except argparse.ArgumentTypeError as exp:
        raise RuntimeError("Error reading input arguments") from exp

    if not args.rally_filepath.is_file():
        raise RuntimeError(f"Invalid filepath given '{args.config_filepath}'")

    try:
        read_config_file(config_path)
    except configparser.Error as exp:
        raise RuntimeError(
            f"could not read influx db config file '{args.config_filepath}'"
        ) from exp


```

Comment (khalford) [reply OpenStack-Rally-Tester/usr/local/bin/rally_extract_results.py:218]: Moving the arg checking outside of main.

**Review by khalford — CHANGES_REQUESTED** [review]

**Review by meoflynn — CHANGES_REQUESTED** [review]

File: OpenStack-Rally-Tester/usr/local/bin/rally_extract_results.py [hunk 6]
```diff
@@ -4,94 +4,217 @@
...
+    """
+    if not data_string:
+        return
+
+    url = f"http://{host}/write?db={db_name}"
+    response = requests.post(url, data=data_string, auth=auth, timeout=60)
+    response.raise_for_status()
+
+
+def main(user_args: List):
```
[the remark this answers is not in the document]
Comment (anish-mudaraddi) [reply OpenStack-Rally-Tester/usr/local/bin/rally_extract_results.py:196]: no it doesn't return anything - it just writes rally test results to a file

**Comment by anish-mudaraddi** [comment]
I'm still waiting for when Ops is not busy so I can grab some dummy rally output files to use for testing

**Comment by anish-mudaraddi** [comment]
I'm closing this until I write the tests

**Pull request closed without merging**